### ISSUE TITLE
Bounding box for separated links

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#2323]
From Gundula: 
<hr>

I do not agree to the following part of the definition for bounding box:
"Where a component consists of disconnected parts, such as a link that wraps onto multiple lines, each part is considered to have its own bounding box."

With the existing definition, a focus indicator is suggested to fully enclose (with four solid lines) each of the disconnected parts, resulting in the appearance of several focus indicators. Next, the link could be named something like "a more detailed view", the part on the first line being "a". As a result the length of the shortest side is the length of the "a", which does not follow the intention of the SC.

Suggestion:
add "In such a case, the base for further measurement is the concatenation of all boxes for disconnect parts. In the example of a link that wraps onto multiple lines the resulting box would enclose the same link if it would not wrap."

The definition of minimum bounding box implies there is only one minimum bounding box for a given UI element. Consequently, the normative text should say 'the minimum bounding box', not 'a minimum bounding box'.

**Comment by alastc** [comment]
Hi @gundulaniemann, 

> In such a case, the base for further measurement is the concatenation of all boxes for disconnect parts. In the example of a link that wraps onto multiple lines the resulting box would enclose the same link if it would not wrap.

I don't see how that would work, if you evaluate what the overall size requirement is based on a complete link, how do you apply that to each part of a wrapped link?

The common cases in HTML are outline and border, where outline creates separate boxes for each part and border doesn't include one side. (See my [test page](https://alastairc.uk/tests/wcag22-examples/focus-more-visible-6.html), at the bottom.)

So the outline case is ok, the question is whether to make the border case pass/fail.

As it stands the border-case can fail if it is 1px thick, I think that is what you are asking for.

I don't see how the change helps, it makes it harder to understand.

**Comment by alastc** [comment]
Hi Gundula,

Ok, so you're trying to take the approach of treating it as one item and base the calculations on that.

I don't think adding to the definition helps, that part would need to be changed. So where it says:

> Where a component consists of disconnected parts, such as a link that wraps onto multiple lines, each part is considered to have its own bounding box.

That could be:
"Where a component consists of disconnected parts, such as a link that wraps onto multiple lines, the component should be treated as though it is connected and the bounding box encloses the combined component."

Or something like that. I don't have strong feelings about this aspect, the main thing I'm trying to avoid is where the indicator size includes the space between parts of the link. This is what you'd get by drawing a bounding box around the separated parts:
<img src="https://user-images.githubusercontent.com/216630/165986704-b0181b39-5610-4d74-990f-e27af8a55c38.png" width="300" alt="A paragraph of text with a link wrapped from the top right to the bottom left. A large red square encloses both parts of the link, much bigger than the size of the link itself.">

**Comment by mbgower** [comment]
First, let me tackle the frankenstein component I [described earlier](https://github.com/w3c/wcag/issues/2323#issuecomment-1112274098)
Imagine a square object

<img width="207" alt="Screenshot 2022-04-29 at 12 59 33 PM" src="https://user-images.githubusercontent.com/14298245/166061533-cf3bb633-9062-4585-891b-50d12665afc0.png">

Then we put a deep blue focus indicator around it
<img width="214" alt="image" src="https://user-images.githubusercontent.com/14298245/166062318-c8277e86-a556-41c2-9c4c-1c996ad7ed87.png">

Now, let's imagine for whatever reason, the initial object is actually made up of 4 smaller squares that are disconnected parts. If those are right together, a single focus indicator might seem to make sense.

<img width="222" alt="image" src="https://user-images.githubusercontent.com/14298245/166062549-ab223780-ea9f-4c48-b471-c6058047f275.png">

But what if the disconnect parts get pulled apart, so that each square's inner corner starts where the outer corner originally began...

<img width="373" alt="image" src="https://user-images.githubusercontent.com/14298245/166062885-d9cb46d5-6082-4dbe-8114-f6c6526cbbc5.png">

With the proposed rewording, a focus rectangle where each of the disconnected pieces only has half a focus indicator would be acceptable (and I'm doing very rough creation here, so excuse the obvious glitches in spacing, etc)

<img width="374" alt="image" src="https://user-images.githubusercontent.com/14298245/166064266-ad3a02ca-7812-4b5c-b2d7-8d2093e8822e.png">

Again, I can't imagine a construct like this occurring, but it's useful to understand what could be allowed from the proposed wording. As it stands right now, each would need its own enclosing focus indicator, as I roughly picture here with a blue focus square around each square:

<img width="372" alt="image" src="https://user-images.githubusercontent.com/14298245/166065756-c99b222a-c4c7-4a54-ae77-86396080093b.png">

I think the current wording leads to a better focus indicator.

**Comment by mbgower** [comment]
> the component should be treated as though it is connected and the bounding box encloses the combined component."

For links, the problem is that the parts of the component are not in the same xy space.

 I'm wondering if the easier solution here could be to _except_ text links, which are fairly unique in that the location shifts unpredictably based on text sizing and responsive design compounding factors.
I cannot think of another component where the same result occurs. Can anyone? If not, how about:

But it's actually hard to except links in that way -- in fact i think the existing exception is meant _specifically_ for links. IMO, the amount of contortions we're doing just to allow someone to have a 1px width 3-sided focus indicator on a text link seems unnecessary. It's easy for someone to resolve: use outline OR bump it up to 2px

**Comment by mbgower** [comment]
I have made a small change to the first sub-bullet of the final exception to keep someone from being able to apply the wrong bounding box in a component with disconnected parts.

**Comment by gundulaniemann** [comment]
IN the above example with the línk wrapping over three lines it feels odd that there are vertical indicators while the focus in fact does not end.
I think the measurement may not force to draw that lines. Also, if we see a 2px line as a solution, in fact we enforce the focus indicator to always have 2px width,as a designer can't predict when a link wraps, and the focus indicator visualization should be consistent all over the UI.

Also, the browser standard might draw a joint focus indicator of the wrapped parts are near each other vertically.
![image](https://user-images.githubusercontent.com/57541275/166457108-8c705bf8-d36f-4da3-836d-06e576b42f63.png)
I fee this also should be allowed.

How can we put this into words?

**Comment by alastc** [comment]
> In the above example with the línk wrapping over three lines it feels odd that there are vertical indicators while the focus in fact does not end.

That seems normal to me, it is the default.

> I feel this also should be allowed.

I think it is, it meets the perimeter measure of the primary bullets, and the size of the exception. It is only when it is separated with gaps that it becomes a possible issue.

**Comment by mbgower** [comment]
> Also, the browser standard might draw a joint focus indicator of the wrapped parts are near each other vertically.... I think this should be allowed

That is allowed through the primary bullets. As well, in the case where this is done by the user agent alone, even if it didn't pass the primaries, we have 2 different exceptions for user agents.

**Comment by alastc** [comment]
Based on the [discussion and resolution from today's meeting](https://www.w3.org/2022/05/03-ag-minutes.html#t10) I'm closing this.

The factors considered were:

- This affects people using a CSS border to create a focus indicator, not outline.
- It is unusual to use border for this purpose as it affects layout. For incline links this causes text to visually move around as you tab through a page. 
- It is easily overcome by either increasing the thickness (to 2px), or using outline.

NB: The small normative update was agreed.
